### PR TITLE
refactor(module:hashcode): refactor control flow component

### PR DESCRIPTION
**Pull request by Nicoss54**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Hashcode component still use the older control flow

Issue Number: N/A

## What is the new behavior?

HashCode component now use the new control flow

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: components/hash-code/hash-code.component.ts [hunk 1]
```diff
@@ -263,6 +275,11 @@ export class NzHashCodeComponent implements OnChanges {
     this.cdr.markForCheck();
   }
 
-  protected readonly isTemplateRef = isTemplateRef;
-  protected readonly isNonEmptyString = isNonEmptyString;
+  get isNsLogoTemplateRef(): boolean {
```
Comment (HyperLife1119) [inline components/hash-code/hash-code.component.ts:278]: isNzLogoTemplateRef

Comment (Nicoss54) [reply components/hash-code/hash-code.component.ts:278]: done

File: components/grid/demo/playground.ts [hunk 2]
```diff
@@ -29,12 +29,17 @@ import { NzMarks } from 'ng-zorro-antd/slider';
...
     <div class="gutter-example">
       <div nz-row [nzGutter]="[hGutter, vGutter]">
-        <div nz-col class="gutter-row" [nzSpan]="24 / count" *ngFor="let i of array">
-          <div class="grid-config">Column</div>
-        </div>
-        <div nz-col class="gutter-row" [nzSpan]="24 / count" *ngFor="let i of array">
-          <div class="grid-config">Column</div>
-        </div>
+        @for (i of array; track i) {
+          <div nz-col class="gutter-row" [nzSpan]="24 / count" *ngFor="let i of array">
```
Comment (HyperLife1119) [inline components/grid/demo/playground.ts:33]: Remove the `*ngFor`

File: components/grid/demo/playground.ts [hunk 3]
```diff
@@ -29,12 +29,17 @@ import { NzMarks } from 'ng-zorro-antd/slider';
...
+            <div class="grid-config">Column</div>
+          </div>
+        }
+
+        @for (i of array; track i) {
+          <div nz-col class="gutter-row" [nzSpan]="24 / count" *ngFor="let i of array">
```
Comment (HyperLife1119) [inline components/grid/demo/playground.ts:39]: Remove the `*ngFor`.

Comment (Nicoss54) [reply components/grid/demo/playground.ts]: done

**Review by HyperLife1119 — APPROVED** [review]
LGTM

File: components/grid/demo/playground.ts [hunk 4]
```diff
@@ -30,13 +30,13 @@ import { NzMarks } from 'ng-zorro-antd/slider';
...
             <div class="grid-config">Column</div>
           </div>
         }
 
         @for (i of array; track i) {
-          <div nz-col class="gutter-row" [nzSpan]="24 / count" *ngFor="let i of array">
+          <div nz-col class="gutter-row" [nzSpan]="24 / count">
             <div class="grid-config">Column</div>
           </div>
         }
```
Comment (HyperLife1119) [inline components/grid/demo/playground.ts:42]: I realized that this file is not in the hashcode scope, can you extract this change into a separate PR? :)

Comment (Nicoss54) [reply components/grid/demo/playground.ts:42]: of course :)
